src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -0,0 +1,457 @@
+                    else if (d == double.MaxValue)
+                        display = "double.MaxValue";
+                    else if (d == double.MinValue)
+                        display = "double.MinValue";
+                    else
+                    {
+                        if (display.IndexOf('.') == -1)
+                            display += ".0";
+                        display += "d";
+                    }
+                }
+                else if (arg is float)
+                {
+                    float f = (float)arg;
+
+                    if (float.IsNaN(f))
+                        display = "float.NaN";
+                    else if (float.IsPositiveInfinity(f))
+                        display = "float.PositiveInfinity";
+                    else if (float.IsNegativeInfinity(f))
+                        display = "float.NegativeInfinity";
+                    else if (f == float.MaxValue)
+                        display = "float.MaxValue";
+                    else if (f == float.MinValue)
+                        display = "float.MinValue";
+                    else
+                    {
+                        if (display.IndexOf('.') == -1)
+                            display += ".0";
+                        display += "f";
+                    }
+                }
+                else if (arg is decimal)
+                {
+                    decimal d = (decimal)arg;
+                    if (d == decimal.MinValue)
+                        display = "decimal.MinValue";
+                    else if (d == decimal.MaxValue)
+                        display = "decimal.MaxValue";
+                    else
+                        display += "m";
+                }
+                else if (arg is long)
+                {
+                    long l = (long)arg;
+                    if (l == long.MinValue)
+                        display = "long.MinValue";
+                    else if (l == long.MinValue)
+                        display = "long.MaxValue";
+                    else
+                        display += "L";
+                }
+                else if (arg is ulong)
+                {
+                    ulong ul = (ulong)arg;
+                    if (ul == ulong.MinValue)
+                        display = "ulong.MinValue";
+                    else if (ul == ulong.MinValue)
+                        display = "ulong.MaxValue";
+                    else
+                        display += "UL";
+                }
+                else if (arg is string)
+                {
+                    var str = (string)arg;
+                    bool tooLong = str.Length > stringMax;
+                    int limit = stringMax - THREE_DOTS.Length;
+
+                    StringBuilder sb = new StringBuilder();
+                    sb.Append("\"");
+                    foreach (char c in (string)arg)
+                    {
+                        sb.Append(EscapeControlChar(c));
```

Comment:
AFAIK, they generally end up as <NameOfType>. Admittedly, that's not very useful and we may want to do better in the future, but I don't think they generate anything that needs truncation. To me, truncating ints, doubles, DateTimes, etc. doesn't make much sense.

I think we may want to make some improvements in how we represent arguments. For example, we have an issue that asks us to display the content of arrays, rather than just <System.Int32[]> as we now do. I think we should do that but it will require logic to truncate both the array (in terms of number of elements) and for a string array, the content as well. I'd rather see the general approach established before working on that issue or any others that come our way.